README.adoc

```diff
@@ -6,319 +6,39 @@ ifdef::env-github[:BRANCH: master]
+are elements of the SMART COSMOS core metaphor, and more information on these elements
+can be found here:
 
-== Copyright and License
-Code and documentation copyright 2016 Smartrac Technology Fletcher, Inc.  Code released under link:LICENSE[Apache 2.0 License].
-
-== REST API
-
-*Available methods*
-
-* <<create1, Create - `POST /{type}`>>
-* <<update1, Update - `PUT /{type}/{urn}`>>
-* <<read1, Read - `GET /{type}?fields={keyNames}&page={pageNumber}&size={pageSize}`>>
-* <<read2, Read - `GET /{type}/{urn}?fields={keyNames}`>>
-* <<read3, Read - `POST /{type}/findByUrns?fields={keyNames}`>>
-* <<read4, Read - `POST /{type}/findByKeyValuePairs?page={pageNumber}&size={pageSize}`>>
-* <<delete1, Delete - `DELETE /{type}/{urn}`>>
-
-=== JSON Fields
-[width="100%",options="header"]
-|====================
-| Field | Format | Default | Required | Description
-| `urn` | String representation of a thing identifier in a common scheme, e.g. `urn:thing:uuid:<UUID>` | generated, value and format depend on the persistence layer implementation | | will be generated if not provided
-| `type` | String | | yes |
-| `active` | boolean |  true | no |
-| `tenantUrn` | String representation of the tenant identifier in a common scheme, e.g. `urn:tenant:uuid:<UUID>` |  generated | generated | only in returned JSON Objects
-| [JSON metadata] | JSON |  | false | Metadata for the thing, e.g. `name`, `description`, ..
-|====================
-
-.**URN Scheme**
-Note that the illustrated scheme for URNs is only for documentation purposes. There must not be any assumptions or expectations on the scheme in the REST layer. All URNs or identifiers are just `String` values in the scope of REST modules!
-
-=== URL parameters
-[width="100%",options="header"]
-|====================
-| Parameter | Format | Default | Description
-| `fields` | Comma-separated String list  |  query parameter, not set by default | Metadata to include in the response (all if not set)
-| `page` | String representation long  |  query parameter, defaults to 0 | page number of multi-element returns, starting at 0
-| `size` | String representation of Integer  |  query parameter, defaults 20 | number of elements per page in multi-element returns
-|====================
-
-
-=== API Endpoints
-
-[[create1]]
-==== Create - `POST /{type}`
-
-----
-POST /Building
-----
-
-The fields `name`, `description` and `large` will be added as metadata for this thing.
-
-*Example 1*
-[source,json]
-----
-{
-    "urn": "urn:thing:uuid:346e742e-2f1e-4d91-9ffe-7b38eec6219c",
-    "active": true,
-    "name": "My home",
-    "description": "My home in US",
-    "large": true
-}
-----
-
-.Response
-----
-201 CREATED
-----
-[source,json]
-----
-{
-    "urn": "urn:thing:uuid:346e742e-2f1e-4d91-9ffe-7b38eec6219c",
-    "type": "Building",
-    "tenantUrn": "urn:tenant:uuid:69bb7c6a-a43b-493d-8e9d-e5a3ed65728a",
-    "active": true
-}
-----
+* https://api.smartcosmos.net/microservices/smartcosmos-ext-things-rdao/index.html[Things]
```

Comment:
No argument from me. If Jens says OK, I can do that when I'm fixing the paging tables
in ext-*-rdao.